Automattic/jetpack
# Show Notifications in Admin Bar

This PR is part of a project that has added a dedicated notifications page on Reader.

When a browser doesn't permit third party cookies, the admin bar notifications will not be displayed as they are inside an iframe on another domain, widgets.wp.com.

There is already a check on widgets.wp.com to determine if the browser permits third party cookies. This is available when an iframe loads https://widgets.wp.com/3rd-party-cookie-check/index.html.

The iframe will try to set a cookie, the results of which (either `WPCOM:3PC:blocked` or `WPCOM:3PC:allowed`) are posted back to the parent page.

For non-WPCOM simple sites that do not load the Jetpack Masterbar, the default admin bar is loaded. It currently uses the results to hide the notifications completely if it found that the browser did not support third party cookies. 

This functionality was added in https://github.com/Automattic/jetpack/pull/25448.

This PR removes the hiding and instead allows the changes in admin-bar-v2.js from D106160-code to handle either rendering the iframed notifications or adding redirect to https://wordpress.com/read/notifications

## Proposed changes:
* Removes javascript that hides the notifications on standard WP admin bar if browser doesn't permit third party cookies


## Jetpack product discussion
Project thread can be found at pe7F0s-rZ-p2

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:

* Create a JN site using Jetpack Live Branches below or open [this link](https://jurassic.ninja/create?jetpack-beta&branches.jetpack=add/cookie-checker-iframe&wp-debug-log)
* Connect Jetpack
* In your browser settings enable All or Third-party cookies, e.g. in Chrome enter this in the address bar `chrome://settings/cookies` to goto cookies related settings
* Goto wp-admin of your JN site
* Confirm that you see the WPCOM notifications button in the admin bar
* Click on the button
* Confirm that the notifications load correctly in the iframe
* Now in your browser settings, Block third-party cookies.
On firefox;
<img width="933" alt="Screenshot 2023-03-29 at 15 50 55" src="https://user-images.githubusercontent.com/5560595/228578605-280c4b50-afae-435a-b44c-54b70769bd78.png">

* Reload wp-admin of the JN site
* Confirm that you still see the WPCOM notifications button in the admin bar
* Confirm that WPCOM notifications button is still visible
* Click on the button
* Confirm that you are redirected to https://wordpress.com/read/notifications
